core/src/main/scala/kafka/coordinator/GroupMetadataManager.scala

```diff
@@ -437,11 +438,21 @@ class GroupMetadataManager(val brokerId: Int,
 
         while (currOffset < highWaterMark && !shuttingDown.get()) {
           buffer.clear()
-          val fileRecords = log.read(currOffset, config.loadBufferSize, maxOffset = None, minOneMessage = true)
-            .records.asInstanceOf[FileRecords]
-          val bufferRead = fileRecords.readInto(buffer, 0)
+          // We only add the READ_UNCOMMITTED parameter here because EasyMock in the GroupMetadataManagerTest
+          // cannot seem to deal with the default value for isolationLevel in log.read() -- it throws an
+          // IllegalStateException.
+
+          val fetchDataInfo = log.read(currOffset, config.loadBufferSize, maxOffset = None, minOneMessage = true,
+            isolationLevel = IsolationLevel.READ_UNCOMMITTED)
+
+          val memRecords = fetchDataInfo.records match {
+            case records: MemoryRecords => records
```

Comment:
Sure, we can push allocation into the other branch. This is technically a bug fix, by the way, since `Log.read` can return `MemoryRecords.EMPTY`.